policy-test/tests/api.rs

```diff
@@ -0,0 +1,215 @@
+        let config = rx
+            .next()
+            .await
+            .expect("watch must not fail")
+            .expect("watch must return an updated config");
+        tracing::trace!(?config);
+        assert_eq!(
+            config.protocol,
+            Some(grpc::inbound::ProxyProtocol {
+                kind: Some(grpc::inbound::proxy_protocol::Kind::Http1(
+                    grpc::inbound::proxy_protocol::Http1::default()
+                )),
+            }),
+        );
+        assert_eq!(
+            config.authorizations.first().unwrap().labels,
+            Some(("name".to_string(), "all-admin".to_string()))
+                .into_iter()
+                .collect()
+        );
+        assert_eq!(
+            *config
+                .authorizations
+                .first()
+                .unwrap()
+                .authentication
+                .as_ref()
+                .unwrap(),
+            grpc::inbound::Authn {
+                permit: Some(grpc::inbound::authn::Permit::Unauthenticated(
+                    grpc::inbound::authn::PermitUnauthenticated {}
+                )),
+            }
+        );
+        assert_eq!(
+            config.labels,
+            Some(("name".to_string(), "linkerd-admin".to_string()))
+                .into_iter()
+                .collect()
+        );
+
+        // Delete the `Server` and ensure that the update reverts to the
+        // default.
+        kube::Api::<k8s::policy::Server>::namespaced(client.clone(), &ns)
+            .delete(&server.name(), &kube::api::DeleteParams::default())
+            .await
+            .expect("Server must be deleted");
+        let config = rx
+            .next()
+            .await
+            .expect("watch must not fail")
+            .expect("watch must return an updated config");
+        tracing::trace!(?config);
+        assert_is_default_deny!(config);
+        assert_protocol_detect!(config);
+    })
+    .await;
+}
+
+async fn create_pod(client: &kube::Client, pod: k8s::Pod) -> Result<k8s::Pod> {
+    let api = kube::Api::<k8s::Pod>::namespaced(client.clone(), &pod.namespace().unwrap());
+    let pod = api.create(&kube::api::PostParams::default(), &pod).await?;
+    time::timeout(
+        time::Duration::from_secs(60),
+        await_condition(api, &pod.name(), conditions::is_pod_running()),
+    )
+    .await??;
+    Ok(pod)
+}
+
+fn mk_pause(ns: &str, name: &str) -> k8s::Pod {
+    k8s::Pod {
+        metadata: k8s::ObjectMeta {
+            namespace: Some(ns.to_string()),
+            name: Some(name.to_string()),
+            annotations: Some(
+                vec![
+                    ("linkerd.io/inject".to_string(), "enabled".to_string()),
+                    (
+                        "config.linkerd.io/default-inbound-policy".to_string(),
+                        "deny".to_string(),
+                    ),
+                ]
+                .into_iter()
+                .collect(),
```

Comment:
wonder if it's worth having a `map!` macro or something that does this, if it's going to show up in more tests...